### PR TITLE
Update base template with latest hooks changes & disable gh-actions, precommit hook

### PR DESCRIPTION
**Pull request by technophile-04**
### Description : 
1. Updated to generate deploy from inside we recently fixed it because ppl were having issue on Windows ( [#362 on se-2](https://github.com/scaffold-eth/scaffold-eth-2/pull/368) ) in the commit https://github.com/scaffold-eth/se-2-challenges/pull/2/commits/859e75fa03f3dfa7c1b10fe3a6c448c9b5111d6b
&nbsp;
2. Updated all the hooks to their latest version and also updated `contracts.ts` since we recently handled a beginner-friendly case on  ( [#372 in SE-2](https://github.com/scaffold-eth/scaffold-eth-2/pull/372) ) in commit [96282fc](https://github.com/scaffold-eth/se-2-challenges/pull/2/commits/96282fcb6b7891de1cf0231028fbe1ecd0911934)
&nbsp;
3. disabled precommit and gh-actions in [6c9947e](https://github.com/scaffold-eth/se-2-challenges/pull/2/commits/6c9947ecab1f0dfd964c6af7e609664a3d7b71f0) & [6c9947e](https://github.com/scaffold-eth/se-2-challenges/pull/2/commits/6c9947ecab1f0dfd964c6af7e609664a3d7b71f0)
&nbsp;

----

### EslintConfig : 
So basically we mainly rules coming from [typecipt-eslint](https://typescript-eslint.io/) and [eslint-plugin-next](https://www.npmjs.com/package/@next/eslint-plugin-next). 

The default rules set up [@typescript-eslint/recommended](https://typescript-eslint.io/) are checkout -> [default  @typescript-eslint/recommended config](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/configs/recommended.ts) for their details checkout -> [typescript-eslint rules detail](https://typescript-eslint.io/rules/). 

I went through each of default config but I think we have already overridden the necessary defaults to make it beginner friendly the only update I did here was making `no-unused-vars : warn` instead of `error`

Regarding [eslint-plugin-next](https://www.npmjs.com/package/@next/eslint-plugin-next) checkout out defaults rules sets here -> [default eslint-plugin-next rules](https://nextjs.org/docs/pages/building-your-application/configuring/eslint#eslint-plugin)
I think the defaults setup by plugin are sensible and might dont cause any harm to beginners. The `no-img-element` also gives just a warning which I think is fine. 

### TsConfig : 
We have `strict` enabled in typescript and the default sets by `strict` are : 
- [noImplicitAny](https://www.typescriptlang.org/tsconfig#noImplicitAny) 

- [noImplicitThis](https://www.typescriptlang.org/tsconfig#noImplicitThis)

- [strictNullChecks](https://www.typescriptlang.org/tsconfig#strictNullChecks)

- [strictFunctionTypes](https://www.typescriptlang.org/tsconfig#strictFunctionTypes)

- [strictBindCallApply](https://www.typescriptlang.org/tsconfig#strictBindCallApply)

- [strictPropertyInitialization](https://www.typescriptlang.org/tsconfig#strictPropertyInitialization)

- [noImplicitReturns](https://www.typescriptlang.org/tsconfig#noImplicitReturns)

- [noFallthroughCasesInSwitch](https://www.typescriptlang.org/tsconfig#noFallthroughCasesInSwitch)

- [esModuleInterop](https://www.typescriptlang.org/tsconfig#esModuleInterop)

- [forceConsistentCasingInFileNames](https://www.typescriptlang.org/tsconfig#forceConsistentCasingInFileNames)

Please feel free to suggest enabling/disable any default both for eslint and tsconfig 🙌

---

### ToDo's : 
- [x] Add block explorer 
- [x] Add MetaHeader tags → https://github.com/scaffold-eth/scaffold-eth-2/pull/359/files

cc @carletex (Sorry for some reason I wasn't able to add reviewers 😅)

Note: Actual commits start from [859e75f](https://github.com/scaffold-eth/se-2-challenges/pull/2/commits/859e75fa03f3dfa7c1b10fe3a6c448c9b5111d6b) (after the merge)...I think it's due to rebased and merged the previous PR

### CODE REVIEW

**Comment by technophile-04** [comment]
So after playing with different options I have dropped the idea of disabling [noImplicitAny](https://www.typescriptlang.org/tsconfig#noImplicitAny) keeping tsConfig as it is. 

TLDR; 
If we want to disable [noImplicitAny](https://www.typescriptlang.org/tsconfig#noImplicitAny) we have too disable [strictNullChecks](https://www.typescriptlang.org/tsconfig#strictNullChecks) because if their values are different then the experience of beginner is ruined 

So if we are disabling  [strictNullChecks](https://www.typescriptlang.org/tsconfig#strictNullChecks) and then only its worth disabling [noImplicitAny](https://www.typescriptlang.org/tsconfig#noImplicitAny), should we disbale [strictNullChecks](https://www.typescriptlang.org/tsconfig#strictNullChecks) (I am kind of confused 😅 )? 

checkout -> https://github.com/Microsoft/TypeScript/issues/21015#issuecomment-355440112

--- 

### To try yourself  : 
Just add in tsConfig under `compilerOptions`: 

```
noImplicitAny : false, 
```

And then run `yarn next:check-types` -> this generates some types error which we don't get at `SE-2 main` branch.  And this types error might frustrate `begginer`

if both `noImplicitAny` & `strictNullChecks` are set to false then we won't get any errors. 

So should we disable `strictNullChecks`?

**Comment by carletex** [comment]
This is great job @technophile-04 . Thank you!!!

Regarding the Types... I'd say, let's merge this as it is, branch it for challenge 0, build, and see how it feels
